backend/oqtopus_cloud/provider/routers/jobs.py

```diff
@@ -39,42 +40,73 @@
 
 @router.get(
     "/jobs",
-    response_model=list[JobDef],
+    response_model=list[Union[JobDef, GetJobsResponse]],
+    responses={500: {"model": Detail}},
 )
 @tracer.capture_method
 def get_jobs(
     device_id: str,
+    fields: Optional[str] = None,
     status: Optional[str] = None,
     max_results: Optional[int] = None,
     timestamp: Optional[str] = None,
     db: Session = Depends(get_db),
-) -> list[JobDef]:
+) -> list[Union[JobDef, GetJobsResponse]] | ErrorResponse:
     logger.info("invoked get_jobs")
-    query = db.query(Job).filter(Job.device_id == device_id)
-    if status is not None:
-        query = query.filter(Job.status == status)
-    if timestamp is not None:
-        time = datetime.fromisoformat(timestamp).astimezone(jst)
-        query = query.filter(Job.created_at > time)
-    if max_results is not None:
-        query = query.limit(max_results)
-    models = query.all()
-    jobs: list[JobDef] = []
-    for model in models:
-        job = model_to_schema(model)
-        if isinstance(job, ValueError):
-            logger.warning(str(job))
+    # Fields Control
+    try:
+        if fields is not None:
+            fields_list = fields.split(",")
+            MAP_SCHEMA_TO_MODEL = {v: k for k, v in MAP_MODEL_TO_SCHEMA.items()}
+            converted_fields_list = [
+                MAP_SCHEMA_TO_MODEL[field] for field in fields_list
+            ]
+            columns = [getattr(Job, field) for field in converted_fields_list]
+            arg_select = columns
+        else:
+            arg_select = [Job]
+
+        stmt = select(*arg_select).filter(Job.device_id == device_id)
```

Comment:
Before executing the query, you should ensure that `arg_select` does not contain duplicate columns.
If the returned rows also contain duplicate columns, it indicates that the returned object does not belong to any of `Job`, `GetJobResponse`, or `ValueError`, violating the return type constraint!

Reply:
https://github.com/oqtopus-team/oqtopus-cloud/blob/95b0788414394d812526f434c26ecd95285cd793/backend/oqtopus_cloud/user/routers/jobs.py#L94-L97
Remove duplicate list elements by converting them to dict objects